docs/proposals/20210716-async-azure-resource-creation-deletion.md

```diff
@@ -0,0 +1,365 @@
+    // BastionHostFailedCondition means the bastion host failed to be created.
+    BastionHostFailedCondition clusterv1.ConditionType = "BastionHostFailed"
+
+    // InboundNATRulesReadyCondition means the inbound NAT rules exist and are ready to be used.
+    InboundNATRulesReadyCondition clusterv1.ConditionType = "InboundNATRulesReady"
+    // InboundNATRulesCreatingReason means the inbound NAT rules are being created.
+    InboundNATRulesCreatingReason clusterv1.ConditionType = "InboundNATRulesCreating"
+    // InboundNATRulesDeletingReason means the inbound NAT rules are being deleted.
+    InboundNATRulesDeletingReason clusterv1.ConditionType = "InboundNATRulesDeleting"
+    // InboundNATRulesFailedCondition means the inbound NAT rules failed to be created.
+    InboundNATRulesFailedCondition clusterv1.ConditionType = "InboundNATRulesFailed"
+
+    // AvailabilitySetReadyCondition means the availability set exists and is ready to be used.
+    AvailabilitySetReadyCondition clusterv1.ConditionType = "AvailabilitySetReady"
+    // AvailabilitySetCreatingReason means the availability set is being created.
+    AvailabilitySetCreatingReason clusterv1.ConditionType = "AvailabilitySetCreating"
+    // AvailabilitySetDeletingReason means the availability set is being deleted.
+    AvailabilitySetDeletingReason clusterv1.ConditionType = "AvailabilitySetDeleting"
+    // AvailabilitySetFailedCondition means the availability set failed to be created.
+    AvailabilitySetFailedCondition clusterv1.ConditionType = "AvailabilitySetFailed"
+
+    // RoleAssignmentReadyCondition means the role assignment exists and is ready to be used.
+    RoleAssignmentReadyCondition clusterv1.ConditionType = "RoleAssignmentReady"
+    // RoleAssignmentCreatingReason means the role assignment is being created.
+    RoleAssignmentCreatingReason clusterv1.ConditionType = "RoleAssignmentCreating"
+    // RoleAssignmentDeletingReason means the role assignment is being deleted.
+    RoleAssignmentDeletingReason clusterv1.ConditionType = "RoleAssignmentDeleting"
+    // RoleAssignmentFailedCondition means the role assignment failed to be created.
+    RoleAssignmentFailedCondition clusterv1.ConditionType = "RoleAssignmentFailed"
+)
+```
+
+#### <a name='OpenQuestions'></a>Open Questions
+
+##### <a name='OpenQuestion1'></a>1. What would the scope of the timeout be? What should the timeout duration be?
+
+The 5s timeout, which is quite aggressive, would be per operation (each call). Then we'd have another overall reconcile loop timeout, which already exists. Currently, that timeout is set at 90 minutes (which is also the ARM deployment timeout). With this proposal, we could bring that overall timeout down to something more responsive, such as 30 seconds.
+
+The specific numbers are not set in stone, and should be revised after doing some performance testing with different values and calculating the P99 expected completion time of operations that are not long-running.
```

Comment:
I'm expecting the numbers to become more aggressive with some testing, maybe closer to 2s per call / 15 seconds overall